Add wake lock tests

Honry opened the pull request:
- Tests converted from [chromium layout test suites](https://chromium.googlesource.com/chromium/src.git/+/master/third_party/WebKit/LayoutTests/wake_lock)
 - Test for monitoring 'activechange' event
 - Tests for different wake lock type

https://github.com/w3c/wake-lock/issues/111

<!-- Reviewable:start -->

<!-- Reviewable:end -->

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20078)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/295623048)

Honry commented:
@marcoscaceres, @andrey-logvinov, PTAL.

Honry committed:
Add wake lock tests

 - Tests converted from chromium layout test suites
 - Test for monitoring 'activechange' event
 - Tests for different wake lock type

marcoscaceres commented on wake-lock/wakelock-api.html:
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Wake Lock API Test</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
```
There will probably be quite a few people authoring this and will likely confuse people. Please add a C0 or W3C license to this instead.

marcoscaceres commented on wake-lock/wakelock-api.html:
```diff
@@ -0,0 +1,21 @@
+
+<script>
+
+promise_test(t => {
```
nit: make async/await

marcoscaceres commented on wake-lock/wakelock-object-is-independent.html:
```diff
@@ -0,0 +1,91 @@
+
+<script>
+
+function make_script() {
```
I'd suggest putting this into an innert `<script type="plain/text">` or make this an ES6 template. You can even have this as a plain function, and then just call `f.toString()`, and it will give you the source.

marcoscaceres commented on wake-lock/wakelock-object-is-independent.html:
```diff
@@ -0,0 +1,91 @@
+function make_script() {
+  return '<script>' +
+         '  let iframeWakeLock, iframeRequest;' +
+         '  window.onmessage = message => {' +
```
nit: make this an async function, add await.

marcoscaceres commented on wake-lock/wakelock-object-is-independent.html:
```diff
@@ -0,0 +1,91 @@
+  return '<script>' +
+         '  let iframeWakeLock, iframeRequest;' +
+         '  window.onmessage = message => {' +
+         '    if (message.data === "INIT") {' +
```
Please use a switch statement for `data`

marcoscaceres commented on wake-lock/wakelock-object-is-independent.html:
```diff
@@ -0,0 +1,91 @@
+         '    } else if (message.data === "RELEASED") {' +
+         '      iframeRequest.cancel();' +
+         '      parent.postMessage(iframeWakeLock.active, "*");' +
+         '    }' +
```
Add a default condition, which is an unreached_code error

marcoscaceres commented on wake-lock/wakelock-object-is-independent.html:
```diff
@@ -0,0 +1,91 @@
+    const iframe = document.createElement("iframe");
+    iframe.onload = () => { resolve(iframe); };
+    iframe.srcdoc = src;
+    iframe.style.display = "none";
```
I'd suggest not hiding it - in case it gets stuck, it's easier to debug.

marcoscaceres commented on wake-lock/wakelock-object-is-independent.html:
```diff
@@ -0,0 +1,91 @@
+    iframe.onload = () => { resolve(iframe); };
+    iframe.srcdoc = src;
+    iframe.style.display = "none";
+    document.documentElement.appendChild(iframe);
```
`document.body.appendChild()`... adding it to the HTML element doesn't make sense.

marcoscaceres commented on wake-lock/wakelock-object-is-independent.html:
```diff
@@ -0,0 +1,91 @@
+  let request1 = wakeLock1.createRequest();
+  let wakeLock2 = await navigator.getWakeLock("screen");
+  let iframe, request2;
+  return load_iframe(make_script())
```
await each step here.

marcoscaceres commented on wake-lock/wakelock-object-is-independent.html:
```diff
@@ -0,0 +1,91 @@
+    iframe.contentWindow.postMessage('INIT', '*');
+    return wait_for_message(iframe);
+  })
+  .then(message => {
```
Nit: the word `message` makes it sound like it's going to be a string. Maybe rename it to reflect that it's a boolean.

marcoscaceres commented on wake-lock/wakelock-onactivechange.html:
```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Wake Lock 'onactivechange' Test</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
```
As above.

marcoscaceres commented on wake-lock/wakelock-onactivechange.html:
```diff
@@ -0,0 +1,34 @@
+
+<script>
+
+promise_test(t => {
```
async

marcoscaceres commented on wake-lock/wakelock-onactivechange.html:
```diff
@@ -0,0 +1,34 @@
+<script>
+
+promise_test(t => {
+  let request, wakelock, eventWatcher;
```
declare these in place, not at the top. Should also be const.

marcoscaceres commented on wake-lock/wakelock-onactivechange.html:
```diff
@@ -0,0 +1,34 @@
+
+promise_test(t => {
+  let request, wakelock, eventWatcher;
+  return navigator.getWakeLock("screen")
```
As above, await here:
```
const wakeLock = await  navigator.getWakeLock("screen");
```

marcoscaceres commented on wake-lock/wakelock-onactivechange.html:
```diff
@@ -0,0 +1,34 @@
+  .then(wakeLock => {
+    wakelock = wakeLock;
+    assert_false(wakelock.active, "the active is false before wake lock is acquired");
+    eventWatcher = new EventWatcher(t, wakelock, 'activechange');
```
Which script introduces EventWatcher? Never seen that. Maybe add a comment?

Honry replied:
[testharness.js]( http://web-platform-tests.org/writing-tests/testharness-api.html)

marcoscaceres commented on wake-lock/wakelock-onactivechange.html:
```diff
@@ -0,0 +1,34 @@
+  })
+  .then(evt => {
+    assert_true(evt.isTrusted && !evt.bubbles && !evt.cancelable && evt instanceof Event, 'a simple event is fired');
+    assert_equals(evt.type, "activechange", 'the event name is "activechange"');
```
nit: `"` vs `'`... linter will get upset.

marcoscaceres commented on wake-lock/wakelock-type.html:
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>WakeLockType Test</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
```
as above.

marcoscaceres commented on wake-lock/wakelock-type.html:
```diff
@@ -0,0 +1,33 @@
+
+<script>
+
+promise_test(t => {
```
async

marcoscaceres commented on wake-lock/wakelock-type.html:
```diff
@@ -0,0 +1,33 @@
+  });
+}, "Test that wakeLock.type is 'system' when system wake lock is invoked");
+
+promise_test(t => {
```
async

marcoscaceres commented on wake-lock/wakelock-type.html:
```diff
@@ -0,0 +1,33 @@
+}, "'WakeLockTypeNotSupported' is thrown when set an empty wake lock type");
+
+promise_test(t => {
+  return promise_rejects(t, new DOMException('', 'WakeLockTypeNotSupported'), navigator.getWakeLock("unsupported"));
```
as above

marcoscaceres reviewed (changes requested):
Generally, the functionality being tested is great... a few nits and maintainability issues, but looking good.

Honry commented:
@marcoscaceres, many thanks for your exhaustive comments, please check again. :-)

marcoscaceres reviewed (approved):
Just a minor nit: remove the link@rel=author tag.

Honry commented:
@marcoscaceres, Thank you again. The tags have been removed.
Merging it.

Honry committed:
Address @marcoscaceres' comments

Honry committed:
Add '.https' postfix

Honry committed:
Use unified quote

Honry committed:
Remove link@rel=author tag
